wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginManager.java

```diff
@@ -105,12 +102,15 @@ public ConnectionPluginManager(ConnectionProvider connectionProvider, Connection
     this.connectionWrapper = connectionWrapper;
   }
 
-  public void lock() {
-    lock.lock();
+  public ResourceLock obtain() {
+    return lock.obtain();
   }
 
-  public void unlock() {
-    lock.unlock();
+  /*
+  For testing only
+   */
+  public void closeResourceLock() {
```

Comment:
"releaseLock"?

Reply:
I'm not married to any name. releaseLock is fine, and then we can use acquireLock instead of obtain.